jhove-modules/tiff-hul/src/main/java/edu/harvard/hul/ois/jhove/module/tiff/IFD.java

```diff
@@ -560,78 +560,69 @@ protected Property propertyHeader(String type, List entries)
-            // be any, and if there are, we don't know how they're encoded.
-            if (c < 32 || c > 127) {
-                sb.append(byteToHex(c));
-            }
-            else {
-                sb.append((char) c);
-            }
-        }
-        return sb.toString();
+        String [] asciiStrings = readASCIIArray(count, value);
+        return asciiStrings.length > 0 ? asciiStrings[0] : "";
     }
 
-    /** Reads an array of strings from the TIFF file.
+    /**
+     * Reads an array of null-terminated ASCII strings from the TIFF file.
+     * If there are non-ASCII characters, they're converted to hex as %XX.
      * 
-     *  @param  count  Number of strings to read
-     *  @param  value  Offset from which to read
-     *  
+     * @param  count  Length of ASCII field
+     * @param  value  Offset of ASCII field
      */
     protected String [] readASCIIArray(long count, long value)
         throws IOException
     {
-        _raf.seek(value);
+        final int LOWEST_PRINTABLE_ASCII = 32;
+        final int HIGHEST_PRINTABLE_ASCII = 126;
+        final int HIGHEST_VALID_ASCII = 127;
 
-        int nstrs = 0;
-        List<String> list = new LinkedList<>();
         byte[] buf = new byte[(int) count];
+        _raf.seek(value);
         _raf.read(buf);
-        StringBuffer strbuf = new StringBuffer();
-        for (int i=0; i<count; i++) {
-            int b = buf[i];
+
+        boolean invalidAscii = false;
+        List<String> list = new LinkedList<>();
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < count; i++) {
+            int b = Byte.toUnsignedInt(buf[i]);
             if (b == 0) {
-                list.add(strbuf.toString());
-                strbuf.setLength(0);
+                list.add(sb.toString());
+                sb.setLength(0);
             }
             else {
-                // Escape characters that aren't ASCII. There really shouldn't
-                // be any, and if there are, we don't know how they're encoded.
-                if (b < 32 || b > 127) {
-                    strbuf.append(byteToHex((byte) b));
+                if (b < LOWEST_PRINTABLE_ASCII || b > HIGHEST_PRINTABLE_ASCII) {
+                    sb.append(byteToHex((byte) b));
+                    if (b > HIGHEST_VALID_ASCII) invalidAscii = true;
                 }
                 else {
-                    strbuf.append((char) b);
+                    sb.append((char) b);
                 }
             }
         }
-        /* We can't use ArrayList.toArray because that returns an 
-           Object[], not a String[] ... sigh. */
-        String [] strs = new String[nstrs];
-        ListIterator<String> iter = list.listIterator();
-        for (int i=0; i<nstrs; i++) {
-            strs[i] =  iter.next();
+
+        if (invalidAscii) {
+            _info.setMessage(new ErrorMessage(
+                    MessageConstants.TIFF_HUL_72, value));
+            _info.setValid(false);
```

Comment:
Hi I have mixed feelings with declaring invalid TIFF files just because of non ASCII (but ISO-8859-1) characters.
Indeed, I know, because this control doesn't exist, that we have thousand of images that have "Bibliothèque nationale de  France" in comment but are otherwise perfectly fine.

It would be nice if we could make this configurable like the `byteoffset=true` parameter to allow it to simply be a warning.
A parameter like invalidNonAsciiCharacters with the default value true will be great.

Reply:
@tledoux I've been working on a more generic way of ignoring / downgrading / upgrading errors by id using a configuration file passed at run-time. It might prove a useful addition as it makes reporting more flexible. Will do my best to get the PR out for the first part of next week for review.

Reply:
Ha, yeah, we ourselves have _quite_ a few (somewhat questionable) "©" symbols peppered about our TIFF collections. That was actually the motivator for the addition of this check. If JHOVE had had it earlier we would have picked up on this issue sooner and been able to remedy it with a better alternative.

The TIFF 5 and 6 specs are very clear that these fields should only contain "7-bit ASCII". Rendering software has no way to know what character encoding is being used in these fields if not ASCII, and so no characters outside of the ASCII range can be relied upon to render predictably outside of the software which wrote them. Given that, I think marking such TIFFs as invalid is actually fairly desirable, from a preservation perspective, though I certainly sympathize about now needing to deal with the problems our past mistakes have made for us.

The optional parameter isn't a bad idea, if you'd like to implement it, but it does make me think how much better a more general solution would be, allowing us to selectively suppress _any_ messages using our cool newfangled message IDs! Alas... we currently set our statuses separately from our messages, so until message objects carry their own status modifiers, parameters may be the way to go.

Then again, a part of me also can't help but feel that by providing an explicit option for this check we may be implying that it's safe or reasonable to ignore this incompatibility under certain circumstances, which I don't believe is necessarily _true_. So I'm torn... but feel free to add what options you like :)

Reply:
@carlwilson Just now saw your reply. Awesome to know it's already being worked on :)